studio/components/ui/SchemaSelector.tsx

```diff
@@ -0,0 +1,162 @@
+          <Button type="default" size="tiny" onClick={() => refetchSchemas()}>
+            Reload schemas
+          </Button>
+        </Alert_Shadcn_>
+      )}
+
+      {isSchemasSuccess && (
+        <Popover_Shadcn_ open={open} onOpenChange={setOpen} modal={false}>
+          <PopoverTrigger_Shadcn_ asChild>
+            <Button
+              size={size}
+              type="outline"
+              className={`w-full [&>span]:w-full ${size === 'small' ? 'py-1.5' : ''}`}
+              iconRight={
+                <IconCode className="text-foreground-light rotate-90" strokeWidth={2} size={12} />
+              }
+            >
+              <div className="w-full flex space-x-3 py-0.5">
+                <p className="text-xs text-light">schema</p>
+                <p className="text-xs">{selectedSchemaName}</p>
+              </div>
+            </Button>
+          </PopoverTrigger_Shadcn_>
+          <PopoverContent_Shadcn_ className="p-0 w-64" side="bottom" align="start">
+            <Command_Shadcn_>
+              <CommandInput_Shadcn_ placeholder="Find schema..." />
+              <CommandList_Shadcn_>
+                <CommandEmpty_Shadcn_>No schemas found</CommandEmpty_Shadcn_>
+                <CommandGroup_Shadcn_>
+                  <ScrollArea className={(schemas || []).length > 7 ? 'h-[210px]' : ''}>
+                    {schemas?.map((schema) => (
+                      <CommandItem_Shadcn_
+                        key={schema.id}
+                        className="cursor-pointer flex items-center justify-between space-x-2 w-full"
+                        onSelect={() => {
+                          onSelectSchema(schema.name)
+                          setOpen(false)
+                        }}
+                        onClick={() => {
+                          onSelectSchema(schema.name)
+                          setOpen(false)
+                        }}
+                      >
+                        <span>{schema.name}</span>
+                        {schema.name === selectedSchemaName && (
+                          <IconCheck className="text-brand" strokeWidth={2} />
+                        )}
+                      </CommandItem_Shadcn_>
+                    ))}
+                  </ScrollArea>
+                </CommandGroup_Shadcn_>
+                {onSelectCreateSchema !== undefined && (
+                  <CommandGroup_Shadcn_ className="border-t">
+                    <CommandItem_Shadcn_
+                      className="cursor-pointer flex items-center space-x-2 w-full"
+                      onSelect={() => {
+                        onSelectCreateSchema()
+                        setOpen(false)
+                      }}
+                      onClick={() => {
+                        onSelectCreateSchema()
+                        setOpen(false)
+                      }}
+                    >
+                      <IconPlus />
+                      <span>Create a new schema</span>
```

Comment:
you can use `flex gap-*` and then you don't need to scatter these `<span>` elements in here

Reply:
imo, tailwind gap > space

space classes just add arbitrary margins, gap uses css grid gutters